### ISSUE TITLE
5$: Error if class is named 'await' in async function

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#707]

**Comment by ooblegork** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by ooblegork** [comment]
Could you clarify this, perhaps with a minimal code example?

Is this simply saying that you cannot have a class "await", unless in an async class? Likewise, you cannot have a class "yield", unless in a generator class?

**Comment by strager** [comment]
```js
// legal
class await {}

function f() {
  // legal
  class await {}
}

async function g() {
  // illegal; qljs should report an error
  class await {}
}
```

Also, it turns out classes can't be called `yield` ever.

**Comment by strager** [comment]
> Is this simply saying that you cannot have a class "await", unless in an async class?

No; the opposite. You *can* have a class named "await", unless in an `async` function.

**Comment by ooblegork** [comment]
Thanks, appreciate it. I've been able to make some progress. 

Before I make my pull request, one quick question - when creating a diagnostic type for this particular error, how would you like me to number this? i.e. E0???.

**Comment by strager** [comment]
> how would you like me to number this? i.e. E0707?

The number doesn't matter, really. It just needs to be unique.
